docs/Architectural-Overview.md

```diff
@@ -121,175 +153,172 @@ Its a REST service which provides APIs to stores build wise test results and agg
+* Chaos and destructive testing framework: A sub framework responsible for implementing libraries to generate
+  destructive test scenarios and chaos in Target deployments.
 
-For any parallel test execution request, priority will be system with shared lock. If such a system is not available, then it will check availability of system with exclusive lock. This module allocates the system only if that system is healthy. 
+* Object Store Data Integrity framework: A sub framework which stores s3 client state and provides library to test
+  versioned object Store data integrity interleaved across object store version upgrade and after object store downtime.
 
- 
+* HA Tests: High availability test cases which test Consistency, Availability and Reliability of Object Store.
 
- 
+* Test Client: A separate VM or Container configured with Python Environment and prerequisites Linux and Python site
+  packages to run automated test cases.
 
- 
+# Introduction
 
- 
+Cortx-Test is distributed test execution framework comprising multiple test runners called autobots. These autobots run
+heterogeneous tests of several types (functional, limit, scale and HA tests) on different targets in optimized manner
+and collect test results at a central database (MongoDB). The test types supported are component tests or system level,
+functional, load, security tests, etc.
 
- 
+# Background
 
-## Framework Deployment Flow: 
+For QA certification of an enterprise product, it is always a challenge to execute the test universe within a finite
+timeframe (typically a day) especially since the test universe has thousands of test cases of several types including
+component, system, and performance tests. Job schedulers and CI tools like Jenkins solve this problem to an extent by
+providing functionality to executing multiple tests on different machines, however it is always hard to monitor numerous
+test executions and deploy and manage targets and complete certification of a build within a day. The algorithms and
+methods mentioned in the paper explain design and implementation of test executor framework which solves this problem
+with automatic test configuration management, parallel test execution, reporting and storage efficiency.
 
-This framework can be deployed and used in two ways. In multi autobot setup mode and single autobot setup mode. 
+# Framework Components
 
- 
+The following diagram shows all components/modules present in framework.
 
-## Multi Autobot's setup: 
+* Master autobot: This module reads test execution data from test management tool. From this test execution data, it
```

Comment:
Can you do branding change along with this work?  Introduction of Gliese and corbots.